src/Orleans/Utils/TimerWheel.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.Runtime
+{
+    // Removes the need in timer per item, optimized for singlethreaded consumption. 
+    internal class TimerWheel<T> : IDisposable where T : ITimebound
+    {
+        private const int MaxItemsPerLock = 10;
+        private readonly Queue<T> _entries = new Queue<T>();
```

Comment:
Unfortunately comment about this was lost during timer wheel extracting. On the common path callbacks on average gets completed in order they were queued, the difference may be between local and remote calls, but it doesn't matter much. There's a [check](https://github.com/dotnet/orleans/pull/2159/files#diff-24969f74ffdd4f9fe9cc680c955cc0ceR42) for already fired callback  , thus it will be removed from the queue in tens to hundreds of milliseconds after successful reply. I've considered using weak references, but it adds to much overhead, so it should be an last resort.